app/core/simulation.ts

```diff
@@ -1503,7 +1490,11 @@ export default class Simulation extends Schema implements ISimulation {
       })
     }
 
-    if (this.redPlayer && this.id === this.redPlayer.simulationId) {
+    if (
+      this.redPlayer &&
+      this.id === this.redPlayer.simulationId &&
+      !this.isGhostBattle
```

Comment:
this prevents adding a battle result history when playing ghost. This is normally taken care by redPlayer.simulationId not being attributed to this simulation, but i prefer to double check

Reply:
Indeed, seems to be legacy code